src/index.js

```diff
@@ -1,45 +1,125 @@
+    taskText.classList.toggle('done');
+    item.classList.toggle('item-checked');
 
-    const itemAction = document.createElement('i');
-    itemAction.classList.add('fa-ellipsis-vertical', 'item-action');
+    updateTaskCompletion(checkboxIndex);
 
-    list.appendChild(item);
-    item.appendChild(checkbox);
-    item.appendChild(taskText);
-    item.appendChild(itemAction);
+    const task = TaskStore.getTaskByIndex(checkboxIndex);
+
+    const checkbox = item.querySelector('input');
+
+    const itemAction = item.querySelector('i');
+    itemAction.className = '';
+
+    if (task.completed === true) {
+      taskText.style.textDecoration = 'line-through';
+      checkbox.checked = true;
+      itemAction.classList.add('fa', 'fa-trash-alt');
+    } else {
+      taskText.style.textDecoration = 'none';
+      checkbox.checked = false;
+      const itemDots = item.querySelector('i');
+      itemDots.classList.add('fa', 'fa-ellipsis-v');
+    }
   }
-};
+});
+
+// --DELETE SINGLE TASK--//
+
+document.querySelector('.list').addEventListener('click', (e) => {
+  if (e.target.classList.contains('fa-trash-alt')) {
+    const { index } = e.target.parentElement.firstChild.dataset;
+    const item = document.getElementById(`item-${index}`);
+    TaskStore.removeTask(index);
+    item.remove();
+  }
+});
+
+// -----REMOVE ALL DONE FROM THE LIST-----//
+
+document.querySelector('.clear-button').addEventListener('click', () => {
+  const tasks = TaskStore.getTasks();
+  const doneTasks = Array.from(document.querySelectorAll('.done'));
+  doneTasks.forEach((doneTask) => doneTask.parentElement.remove());
+  const uncompletedTasks = [];
+  for (let i = 0; i < tasks.length; i += 1) {
+    if (tasks[i].completed === false) {
+      uncompletedTasks.push(tasks[i]);
+    }
+  }
+
+  for (let i = 0; i < uncompletedTasks.length; i += 1) {
+    uncompletedTasks[i].index = i;
+  }
+  localStorage.setItem('tasks', JSON.stringify(uncompletedTasks));
+  getTasks();
+});
 
-toDoList(tasks);
\ No newline at end of file
+// -----EDIT-----//
+
+document.querySelector('.list').addEventListener('click', (e) => {
+  if (e.target.classList.contains('task-text')) {
+    const text = e.target;
+    const editInput = document.createElement('input');
+    editInput.type = 'text';
+    editInput.classList.add('edit-input');
+    editInput.value = text.textContent;
+    text.parentElement.classList.add('editing');
+    text.parentElement.replaceChild(editInput, text);
+
+    editInput.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        editInput.parentElement.classList.remove('editing');
+
+        const { index } = e.target.previousSibling.dataset;
+        const task = TaskStore.getTaskByIndex(index);
+        task.description = editInput.value;
+
+        text.textContent = editInput.value;
+        editInput.parentElement.replaceChild(text, editInput);
+
+        TaskStore.updateTask(task);
+      }
+    });
+  }
+});
```

Comment:
- [ ] Your app has strange behavior when I add a new task after editing or removing a task from a list and not reloading the page. Sometimes the new task doesn't have the delete icon when I click on the check box and it checks a previous task too. Kindly make sure that the tasks that are displayed in your app are well connected with the logic behind your app.
-
![Untitled](https://user-images.githubusercontent.com/61919617/184384804-c5ab2e46-9ab8-462a-b268-d7c1b5c87bc8.gif)
-